brave/src/main/java/brave/RealSpan.java

```diff
@@ -12,22 +13,22 @@
   final PendingSpans pendingSpans;
   final MutableSpan state;
   final Clock clock;
-  final SpanReporter spanReporter;
+  final Firehose firehose;
   final ErrorParser errorParser;
   final RealSpanCustomizer customizer;
 
   RealSpan(TraceContext context,
       PendingSpans pendingSpans,
       MutableSpan state,
       Clock clock,
-      SpanReporter spanReporter,
+      SpanReporter firehose,
```

Comment:
I see SpanReporter is a subtype of Firehose but did you intend to not change the type here?

Reply:
added a commit to change names etc. Now the router thing is called FirehoseDispatcher and it doesn't implement Firehose.. makes it easier to see what's doing what.